pkg/provider/cloud/vsphere/provider.go

```diff
@@ -159,22 +85,26 @@ func (v *Provider) InitializeCloudProvider(ctx context.Context, cluster *kuberma
 		// If the user did not specify a folder, we create a own folder for this cluster to improve
 		// the VM management in vCenter
 		clusterFolder := path.Join(rootPath, cluster.Name)
-		if err := createVMFolder(ctx, session, clusterFolder); err != nil {
+		created, err := createVMFolder(ctx, session, clusterFolder)
+		if err != nil {
 			return nil, fmt.Errorf("failed to create the VM folder %q: %w", clusterFolder, err)
 		}
 
-		cluster, err = update(ctx, cluster.Name, func(cluster *kubermaticv1.Cluster) {
-			kuberneteshelper.AddFinalizer(cluster, folderCleanupFinalizer)
-			cluster.Spec.Cloud.VSphere.Folder = clusterFolder
-		})
-		if err != nil {
-			return nil, err
+		if created {
+			cluster, err = update(ctx, cluster.Name, func(cluster *kubermaticv1.Cluster) {
+				kuberneteshelper.AddFinalizer(cluster, folderCleanupFinalizer)
+				cluster.Spec.Cloud.VSphere.Folder = clusterFolder
+			})
+			if err != nil {
+				return nil, err
+			}
 		}
```

Comment:
I really don't get how this is even reachable? take a look at the lines above https://github.com/kubermatic/kubermatic/blob/f890674f0adda5a7e44f70873248cad21ddad092/pkg/provider/cloud/azure/resource_group.go#L46-L48, I don't think that this is called at all because of the condition that I attached in this comment, or am I overlooking something here?

Reply:
Anyway, that's actually a good catch, will include that in my resource created check.

Reply:
> I really don't get how this is even reachable? take a look at the lines above
> 
> https://github.com/kubermatic/kubermatic/blob/f890674f0adda5a7e44f70873248cad21ddad092/pkg/provider/cloud/azure/resource_group.go#L46-L48
> 
> I don't think that this is condition will be triggered at all because of the condition that I attached in this comment, or am I overlooking something here?

Not sure I'm following on this one, the code you linked will return an error from `clients.Groups.Get(ctx, cluster.Spec.Cloud.Azure.ResourceGroup, nil)` if that error was something other than a "not found" response. For example server errors or permission errors. If it was a "not found" error, we want to continue with the code, so it doesn't return early.